base/linalg/lq.jl

```diff
@@ -88,22 +88,25 @@ end
 
 size(A::LQPackedQ) = size(A.factors)
 
-full(A::LQ) = A[:L]*A[:Q]
-#=
-We construct the full eye here, even though it seems ineffecient, because
-every element in the output matrix is a function of all the elements of
-the input matrix. The eye is modified by the elementary reflectors held
-in A, so this is not just an indexing operation. Note that in general
-explicitly constructing Q, rather than using the ldiv or mult methods,
-may be a wasteful allocation.
-=#
-function full{T}(A::LQPackedQ{T}; thin::Bool=true)
+convert(::Type{Matrix}, A::LQ) = convert(Array, A[:L]*A[:Q])
+convert(::Type{Array}, A::LQ) = convert(Matrix, A)
+full(A::LQ) = convert(Array, A)
+
+function convert{T}(::Type{Matrix}, A::LQPackedQ{T}; thin::Bool = true)
```

Comment:
We should think this through carefully instead of putting new API's in `convert` which has happened before but was never really a good idea when it snuck in. These are probably different operations or conversions, we should try to find a good way of differentiating.